README.md

```diff
@@ -47,3 +47,50 @@ cd $(nix-build) && bin/server
 ```
 
 **Note**: The current working directory is important when running the server. The server won't be able to find the clientside part of the app when running the server from some place *other* than the folder with `bin` and `static`. This will prevent the javascript from loading and make the buttons not work. In other news, that's a great way of looking at the part of the app sent by the server.
+
+
+## Developing the project
+
+You can work on the different sub-projects (`client`, `common` and `server`) using `nix-shell` and `cabal`.
+
+`common`:
+
+```bash
+cd common
+nix-shell
+cabal build
+cabal run common-test
+...
+exit
+```
+
+`client`:
+
+```bash
+cd client
+nix-shell
+cabal new-build
```

Comment:
You're right, `new-build` is not necessary. I used the `cabal.project` file + `new-build` to specify the ghcjs compiler but it can be done using a `cabal.config` file + traditional cabal commands.